src/convertToObject.js

```diff
@@ -3,16 +3,43 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arrOfProperties
+    = sourceString.split(';')
+      .map((property) => {
+        if (property.indexOf(':') >= 0) {
+          const convertedProp = property.split(':');
+
+          convertedProp[0] = convertedProp[0].trim();
+          convertedProp[1] = convertedProp[1].trim();
+
+          return convertedProp;
+        }
+      });
+
+  const objOfProperties = arrOfProperties.reduce((prev, prop) => {
+    if (prop === undefined) {
```

Comment:
Instead of checking if 'prop' is undefined, you should filter out the undefined values after the map function. This would make your code cleaner and more efficient.